spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClient.java

```diff
@@ -73,11 +81,20 @@ private void addInstancesToList(List<ServiceInstance> instances, String serviceI
 			requestBuilder.setTags(queryTags);
 		}
 		HealthServicesRequest request = requestBuilder.build();
-
-		Response<List<HealthService>> services = this.client.getHealthServices(serviceId, request);
-
-		for (HealthService service : services.getValue()) {
-			instances.add(new ConsulServiceInstance(service, serviceId));
+		try {
+			Response<List<HealthService>> services = this.client.getHealthServices(serviceId, request);
+
+			for (HealthService service : services.getValue()) {
+				instances.add(new ConsulServiceInstance(service, serviceId));
+			}
+			if (properties.isBackup()) {
+				serviceInstanceBackup.put(serviceId, instances);
```

Comment:
Shouldn't you purge the "backup" before adding the new instances?  You could end up with stale data in the backup.

Reply:
I think that the same serviceId, after being added, the old instances will be covered by new data, isn't it? @ryanjbaxter

Reply:
What if the service was removed?

Reply:
I think that when the service is removed, consul can also accept the request. The data inside the Backup will be replaced by a empty list. What should this be reasonable? But if consul does not provide a service normally, it should return the last acquisition of the service instance list, no matter whether it is empty.

Reply:
Sorry i thought this code was iterating all services and not instances of a given service.

Reply:
Yes, but if your assumption, `instances` will be an empty list when there is no service instance returns, which will also be placed in the map. isn't it?

Reply:
Yes correct, I don't think there is a problem after all